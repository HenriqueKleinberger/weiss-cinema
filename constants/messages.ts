export const ERROR_SEARCHING_MOVIE_DETAILS =
  'Error on searching the movie details.';
export const MOVIE_NOT_FOUND_MESSAGE = 'Movie not found.';
export const REMOVE_FROM_WISHLIST = 'remove from wishlist';
export const ADD_TO_WISHLIST = 'add to wishlist';
export const SEARCH_MOVIE = 'Search a movie.';
export const SEARCH_RESULTS_FOR = 'Search Results For';
export const SEARCH_RESULTS_FOR_ON_WISHLIST = 'Search Results For On Wishlist';
export const EMPTY_WISHLIST =
  'Your wishlist is empty, save new movies of your interest';
