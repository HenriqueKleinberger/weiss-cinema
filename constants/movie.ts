export const DATA_NOT_AVAILABLE = 'N/A';
