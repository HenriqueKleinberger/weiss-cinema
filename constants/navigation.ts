export const HOME = '/';
export const WISHLIST = '/wishlist';
