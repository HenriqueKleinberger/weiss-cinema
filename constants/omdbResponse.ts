export const TRUE = 'True';
export const MOVIE_NOT_FOUND = 'Movie not found!';
