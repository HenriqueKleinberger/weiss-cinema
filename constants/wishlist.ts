export const LOCAL_STORAGE_WISHLIST = 'weiss-cinema-wishlist';
