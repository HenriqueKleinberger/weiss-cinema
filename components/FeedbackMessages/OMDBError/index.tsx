import React from 'react';
import { MOVIE_NOT_FOUND } from '../../../constants/omdbResponse';
import MovieNotFound from './MovieNotFound';
import { Container } from '../styles';

interface IProps {
  message: string;
}

const OMDBError = ({ message }: IProps) => {
  if (message === MOVIE_NOT_FOUND) {
    return <MovieNotFound />;
  }

  return (
    <Container>
      <p>{message || ''}</p>
    </Container>
  );
};

export default OMDBError;
