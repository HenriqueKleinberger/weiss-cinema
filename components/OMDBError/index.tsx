import React from 'react';
import { MOVIE_NOT_FOUND } from '../../constants/omdbResponse';
import MovieNotFound from './MovieNotFound';

const OMDBError = ({ errorMessage }: { errorMessage: string }) => {
  if (errorMessage === MOVIE_NOT_FOUND) {
    return <MovieNotFound />;
  }

  return <div>{errorMessage || 'Buscar um filme'}</div>;
};

export default OMDBError;
