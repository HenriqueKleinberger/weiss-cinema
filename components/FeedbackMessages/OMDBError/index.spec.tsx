import React from 'react';
import { shallow } from 'enzyme';
import OMDBError from './index';
import { MOVIE_NOT_FOUND } from '../../../constants/omdbResponse';

describe('OMDBError', () => {
  it('should match snapshot when is message is generic', () => {
    const wrapper = shallow(<OMDBError message="Error" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should return MovieNotFound component when message equals movie not found', () => {
    const wrapper = shallow(<OMDBError message={MOVIE_NOT_FOUND} />);
    expect(wrapper.find('MovieNotFound').length).toBe(1);
  });
});
