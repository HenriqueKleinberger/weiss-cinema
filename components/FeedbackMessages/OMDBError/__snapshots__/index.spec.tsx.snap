// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OMDBError should match snapshot when is message is generic 1`] = `
<styled.div>
  <p>
    Error
  </p>
</styled.div>
`;
